components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -283,6 +288,8 @@ func Run(options ...RunOption) {
 
 		// We need to checkout dotfiles first, because they may be changing the path which affects the IDE.
 		// TODO(cw): provide better feedback if the IDE start fails because of the dotfiles (provide any feedback at all).
+		<-apiReadyCtx.Done()
+		time.Sleep(200 * time.Millisecond)
```

Comment:
What does the sleep do?

`time.Sleep`/`time.After` without comment or clear context are usually a bit of a red flag :)

Reply:
instead of relying on the `git -> credential helper` path, we could also re-use a Git content initializer and feed the token directly from the token service. This way we don't loose time waiting for the services to come up.